gramex/apps/nlg/nlg.js

```diff
@@ -0,0 +1,506 @@
+/* globals currentEditIndex, grammarOptions, templates, args, df, currentEventHandlers, nlg_base */
+/* exported addToNarrative, setInitialConfig, checkTemplate, saveTemplate, addCondition, addName, changeFHSetter, shareNarrative, copyToClipboard */
+/* eslint-disable no-global-assign */
+var narrative_name, dataset_name
+
+class Template {
+  // Class to hold a piece of text that gets rendered as a
+  // tornado template when the narrative is invoked anywhere.
+  constructor(
+    text, tokenmap, inflections, fh_args, condition = '', setFHArgs = false, template = '',
+    previewHTML = '', grmerr = null, name = ''
+  ) {
+    this.source_text = text
+    this.tokenmap = tokenmap
+    this.inflections = inflections
+    for (let [token, tkobj] of Object.entries(tokenmap)) {
+      if (Array.isArray(tkobj)) {
+        this.tokenmap[token] = new Token(this, token, tkobj, this.inflections[token])
+      }
+      else {
+        var newToken = new Token(this, token, tkobj.tokenlist, tkobj.inflections)
+        newToken.template = tkobj.template
+        this.tokenmap[token] = newToken
+      }
+    }
+    this.fh_args = fh_args
+    this.setFHArgs = setFHArgs
+    this.condition = condition
+    this.template = template
+    this.previewHTML = previewHTML
+    this.grmerr = grmerr
+    this.name = name
+  }
+
+  checkGrammar() {
+    // Check the English text for grammatical errors using LanguageTool.
+    // learn.gramener.com/guide/languagetool
+    let self = this
+    $.getJSON(
+        nlg_base + '/languagetool/?lang=en-us&q=' + encodeURIComponent(`${this.source_text}`)
```

Comment:
Since you're using template literals, use for entire url construction -- ``` `${nlg_base}/languagetool/?lang=en-us&q=${encodeURIComponent(this.source_text)}` ```